### PR TITLE
REL-507462 - Updated ImagingHelper.cs RSAPI calls to REST

### PR DESCRIPTION
**Pull request by aarongilbert123**

### CODE REVIEW

**Review by TommyBigNose — APPROVED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/ImagingHelper.cs [hunk 1]
```diff
@@ -158,18 +168,37 @@ private async Task<bool> JobCompletedSuccessfullyAsync(int workspaceArtifactId,
...
 			{
 				while (currentWaitTimeInMilliseconds < maxTimeInMilliseconds && jobComplete == false)
 				{
-					Thread.Sleep(sleepTimeInMilliSeconds);
-
-					RDO job = await Task.Run(() => RsapiClient.Repositories.RDO.ReadSingle(imagingSetArtifactId));
-					jobComplete = job[fieldGuid].ValueAsFixedLengthText.Contains("Completed");
-
-					currentWaitTimeInMilliseconds += sleepTimeInMilliSeconds;
+					HttpClient httpClient = RestHelper.GetHttpClient(InstanceAddress, AdminUsername, AdminPassword);
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/ImagingHelper.cs:175]: Can we use the Retry function from the Polly tool which Tom used in one of the helpers to retry waiting. I believe its in Application Helper class. 
I think that approach is much cleaner and re-usable and better than any custom code we write. Also code will be consistent across our helpers.

- [ ] use the Retry function from the Polly tool

**Review by calimetikcura — CHANGES_REQUESTED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/ImagingHelper.cs (hunk 1 above)
Comment (TommyBigNose) [reply CSharp/DevVmPowershell/Helpers/Implementations/ImagingHelper.cs:175]: Just a note, if the function you're retrying doesn't throw an error if you don't get the desired value, you can quickly wrap that up in a dynamic function to throw one if the value isn't want you wanted.

Comment (aarongilbert123) [reply CSharp/DevVmPowershell/Helpers/Implementations/ImagingHelper.cs]: Updated to use RetryLogicHelper

**Review by TommyBigNose — DISMISSED** [review]

File: CSharp/DevVmPowershell/Helpers/Constants.cs [hunk 2]
```diff
@@ -80,6 +80,8 @@ public class Waiting
 		{
 			public const int MAX_WAIT_TIME_IN_MINUTES = 10;
 			public const int SLEEP_TIME_IN_SECONDS = 15;
+			public const int RETRY_COUNT = 60;
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Constants.cs:83]: - [ ] If these values are specific to the imagingHelper waiting, I would put that in the variable name to be specific

**Review by calimetikcura — CHANGES_REQUESTED** [review]

**Review by calimetikcura — APPROVED** [review]

**Review by TommyBigNose — APPROVED** [review]

**Pull request merged**